src/stream/src/executor/mview/materialize.rs

```diff
@@ -503,6 +531,48 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
                                 }
                             };
                         }
+
+                        ConflictBehavior::DoUpdateIfNotNull => {
+                            match self.force_get(&key) {
+                                Some(old_row) => {
+                                    // In this section, we compare the new row and old row column by column and perform `DoUpdateIfNotNull` replacement.
+                                    // todo(wcy-fdu):find a way to output the resulting new row directly to the downstream chunk, thus avoiding an additional deserialization step.
+                                    let mut old_row_deserialized_vec = row_serde
+                                        .deserializer
+                                        .deserialize(old_row.row.clone())?
+                                        .into_inner()
+                                        .into_vec();
+                                    let new_row_deserialized =
+                                        row_serde.deserializer.deserialize(value.clone())?;
+
+                                    execute_do_update_if_not_null_replacement(
+                                        &mut old_row_deserialized_vec,
+                                        new_row_deserialized,
+                                    );
+                                    let new_row = OwnedRow::new(old_row_deserialized_vec);
+                                    let new_row_bytes =
+                                        Bytes::from(row_serde.serializer.serialize(new_row));
+                                    fixed_changes().update(
+                                        key.clone(),
+                                        old_row.row.clone(),
+                                        new_row_bytes.clone(),
+                                    );
+                                    // update cache
+                                    self.data.push(
+                                        key.clone(),
+                                        Some(CompactedRow { row: new_row_bytes }),
+                                    );
+                                }
+                                None => {
+                                    fixed_changes().insert(key.clone(), value.clone());
+                                    // update cache
+                                    self.data.push(
+                                        key.clone(),
+                                        Some(CompactedRow { row: value.clone() }),
```

Comment:
I think the clone can be omit, but we can find a more elegant way to update the cache later.